File: 프로그래머스/lv0/120866. 안전지대/안전지대.py
def solution(board):
    n = len(board)
    cnt = 0
    for i in range(n):
        for j in range(n):
            if board[i][j] == 1:
                continue
            flag = True
            for d in ((-1,0), (1,0), (0,-1), (0,1), (-1, -1), (-1, 1), (1, 1), (1, -1)):
                nx = i + d[0]
                ny = j + d[1]
                if not (0<= nx < n and 0<= ny < n):
                    continue
                if board[nx][ny] == 1:
                    flag = False
                    break
            if flag:
                cnt += 1
                
    return cnt
            

File: 프로그래머스/0/120866. 안전지대/안전지대.java
class Solution {

    int[] dx = {-1, -1, 0, 1, 1, 1, 0, -1};
    int[] dy = {0, 1, 1, 1, 0, -1, -1, -1};

    public int solution(int[][] board) {
        int danger = 0;

        for (int i = 0; i < board.length; i++) {
            for (int j = 0; j < board.length; j++) {
                if (board[i][j] == 1) {
                    danger += 1;
                    for (int k = 0; k < 8; k++) {
                        int nx = i + dx[k];
                        int ny = j + dy[k];
                        if (nx >= 0 && nx < board.length && ny >= 0 && 
                            ny < board.length && board[nx][ny] == 0) {
                            board[nx][ny] = 2;
                            danger += 1;
                        }
                    }
                }
            }
        }

        return board.length * board.length - danger;
    }
}